gapis/resolve/dependencygraph2/graph_visualization/graph_algorithms.go

```diff
@@ -0,0 +1,142 @@
+// Copyright (C) 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package graph_visualization
+
+import (
+	"github.com/google/gapid/gapis/api"
+)
+
+const (
+	NO_VISITED       = 0
+	VISITED_AND_USED = -1
+	FRAME            = "FRAME"
+	UNUSED           = "UNUSED"
+)
+
+func (g *graph) traverseGraph(currentNode *node, visitTime, minVisitTime, idInStronglyConnectedComponents, visitedNodesId *[]int, currentId, currentTime *int) {
```

Comment:
These are large enough functions, it would be good to get comments on exactly what the parameters are supposed to be.

Reply:
I abstracted the parameters used to compute the Strongly Connected Components i a directed graph based on Tarjan algorithm. Thank you.